contracts/exchange/test/transactions_unit_tests.ts

```diff
@@ -0,0 +1,675 @@
+            const result = await transactionsContract.batchExecuteTransactions.callAsync(
+                [transaction1, transaction2],
+                [randomSignature(), randomSignature()],
+                {
+                    from: accounts[0],
+                },
+            );
+
+            // Create an abiEncoder for bytes. This will be used to decode the result and encode what
+            // is expected.
+            const abiEncoder = AbiEncoder.create('bytes');
+
+            // Ensure that the result contains the abi-encoded bytes "0xdeadbeef"
+            const encodedDeadbeef = abiEncoder.encode('0xdeadbeef');
+            expect(result.length).to.be.eq(2);
+            expect(
+                result[0] ===
+                    '0x0000000000000000000000000000000000000000000000000000000000000020'.concat(
+                        encodedDeadbeef.slice(2, encodedDeadbeef.length),
+                    ),
+            ).to.be.true();
+
+            // Ensure that the result contains the abi-encoded bytes "0xdeadbeef"
+            const encodedBeefdead = abiEncoder.encode('0xbeefdead');
+            expect(result.length).to.be.eq(2);
+            expect(
+                result[1] ===
+                    '0x0000000000000000000000000000000000000000000000000000000000000020'.concat(
+                        encodedBeefdead.slice(2, encodedBeefdead.length),
+                    ),
+            ).to.be.true();
+
+            // Verify that the logs returned from the call are correct.
+            const receipt = await logDecoder.getTxWithDecodedLogsAsync(
+                await transactionsContract.batchExecuteTransactions.sendTransactionAsync(
+                    [transaction1, transaction2],
+                    [randomSignature(), randomSignature()],
+                    {
+                        from: accounts[0],
+                    },
+                ),
+            );
+
+            // Verify that the correct number of events were logged.
+            const logs = receipt.logs as Array<LogWithDecodedArgs<TestTransactionsTransactionExecutionEventArgs>>;
+            expect(logs.length).to.be.eq(4);
+
+            // Ensure that the correct events were logged.
+            expect(logs[0].event).to.be.eq('ExecutableCalled');
+            expect(logs[0].args.data).to.be.eq(constants.NULL_BYTES);
+            expect(logs[0].args.returnData).to.be.eq('0xdeadbeef');
+            expect(logs[1].event).to.be.eq('TransactionExecution');
+            expect(logs[1].args.transactionHash).to.eq(transactionHash1);
+            expect(logs[2].event).to.be.eq('ExecutableCalled');
+            expect(logs[2].args.data).to.be.eq(constants.NULL_BYTES);
+            expect(logs[2].args.returnData).to.be.eq('0xbeefdead');
+            expect(logs[3].event).to.be.eq('TransactionExecution');
+            expect(logs[3].args.transactionHash).to.eq(transactionHash2);
+        });
+    });
```

Comment:
This should pass. Why is it an edge case?

Reply:
Because it's an ultra specific scenario that takes advantage of all the reentrancy and context management features in the system. Basically, if we changed any one of them this would probably fail so it might be a good way to check for regressions.

Reply:
@dorothy-zbornak Do you think it would make sense to add a simpler test for this? In this case, we could nest in an equally valid way by excluding the `batchExecuteTransaction` and just adding a single call to `executeTransaction`. Not to say that we shouldn't include the test case that you describe, but I think it would also be good to have a super simple case.

Reply:
Yeah, we could probably afford to peel off one layer. I think all the other stuff kind of has to go together to hit all the paths at once, though. :thinking: I do think we're pretty well covered for now.